app/src/main/java/com/hedvig/app/feature/offer/ui/OfferActivity.kt

```diff
@@ -127,20 +132,27 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
                     ConcatSpanSizeLookup(gridLayoutManager.spanCount) { concatAdapter.adapters }
             }
 
-            model.viewState.observe(this@OfferActivity) { viewState ->
-                when (viewState) {
-                    OfferViewModel.ViewState.HasContracts -> startLoggedInActivity()
-                    is OfferViewModel.ViewState.OfferItems -> {
-                        topOfferAdapter.submitList(viewState.topOfferItems)
-                        perilsAdapter.submitList(viewState.perils)
-                        insurableLimitsAdapter.submitList(viewState.insurableLimitsItems)
-                        documentAdapter.submitList(viewState.documents)
-                        bottomOfferAdapter.submitList(viewState.bottomOfferItems)
+            lifecycleScope.launchWhenResumed {
+                model.viewState.collect { viewState ->
+                    when (viewState) {
+                        OfferViewModel.ViewState.HasContracts -> startLoggedInActivity()
+                        is OfferViewModel.ViewState.OfferItems -> {
+                            topOfferAdapter.submitList(viewState.topOfferItems)
+                            // Workaround - recyclerview will scroll to bottom if updating all items simultaneously.
+                            lifecycleScope.launch {
+                                delay(500)
```

Comment:
Interesting 🤔 is 500 ms required, or just a number that works?

I guess as long as we do it after the `RecyclerView` has finished laying out the new items, we should be good, right? Perhaps we could use `doOnNextLayout` and post the additional items from that 🙂

Reply:
Good idea, that works great.